File: baselibrary/src/main/java/com/minicart/android/baselibrary/mvp/IView.java
package com.minicart.android.baselibrary.mvp;

import android.content.Context;
import android.content.Intent;

/**
 * @类名：IView
 * @描述：
 * @创建人：54506
 * @创建时间：2017/1/13 11:04
 * @版本：
 */

public interface IView<P extends IPresenter> {

    P getPresenter();

    Context getContext();

    void showLoadingMessage(String message);

    void updateLoadingMessage(String msg);

    void hideLoading();

    void hideLoading(long delayedMillisecond);

    void launchActivity(Intent intent);

    void killMyself();
}

File: baselibrary-kotlin/src/main/java/com/minicart/android/baselibrary/mvp/IView.kt
package com.minicart.android.baselibrary.mvp

import android.content.Context
import android.content.Intent

/**
 * @类名：IView
 * *
 * @描述：
 * *
 * @创建人：54506
 * *
 * @创建时间：2017/1/13 11:04
 * *
 * @版本：
 */

interface IView<out P> {

    fun getPresenter(): P?

    fun getContext(): Context

    fun showLoadingMessage(message: String)

    fun updateLoadingMessage(msg: String)

    fun hideLoading()

    fun hideLoading(delayedMillisecond: Long)

    fun launchActivity(intent: Intent)

    fun killMyself()
}
